### PR TITLE
feat: Deploy Factories

### PR DESCRIPTION
**Pull request by prathmeshkhandelwal1**

### CODE REVIEW

File: solidity/scripts/MultichainDeploy.sol [hunk 1]
```diff
@@ -11,11 +11,11 @@ import {ScriptingLibrary} from './ScriptingLibrary/ScriptingLibrary.sol';
 contract MultichainDeploy is Script, ScriptingLibrary {
   uint256 public deployer = vm.envUint('DEPLOYER_PRIVATE_KEY');
   address constant CREATE2 = 0x4e59b44847b379578588920cA78FbF26c0B4956C;
-  string[] public chains = ['POLYGON_RPC', 'OPTIMISM_RPC', 'GOERLI_RPC'];
+  string[] public chains = ['MAINNET_RPC', 'POLYGON_RPC', 'OPTIMISM_RPC', 'ARBITRUM_RPC', 'GNOSIS_RPC', 'BSC_RPC'];
 
   function run() public {
     //TODO: Change salt from this test to prod before release
```
Comment (0xGorilla) [inline solidity/scripts/MultichainDeploy.sol:17]: Please remove this comment

File: solidity/scripts/MultichainDeploy.sol [hunk 2]
```diff
@@ -11,11 +11,11 @@ import {ScriptingLibrary} from './ScriptingLibrary/ScriptingLibrary.sol';
...
-    bytes32 _salt = keccak256(abi.encodePacked('xxxsdsdd23ewXERewewCewew20Factoewewry', msg.sender));
+    bytes32 _salt = keccak256(abi.encodePacked('de4y6D2tuMLufI7YOZMZ9AKw6YCMZeII7mX4U', msg.sender));
```
Comment (0xGorilla) [inline solidity/scripts/MultichainDeploy.sol:18]: How did you generate this salt?

File: solidity/scripts/MultichainDeploy.sol [hunk 3]
```diff
@@ -11,11 +11,11 @@ import {ScriptingLibrary} from './ScriptingLibrary/ScriptingLibrary.sol';
 contract MultichainDeploy is Script, ScriptingLibrary {
   uint256 public deployer = vm.envUint('DEPLOYER_PRIVATE_KEY');
   address constant CREATE2 = 0x4e59b44847b379578588920cA78FbF26c0B4956C;
-  string[] public chains = ['POLYGON_RPC', 'OPTIMISM_RPC', 'GOERLI_RPC'];
+  string[] public chains = ['MAINNET_RPC', 'POLYGON_RPC', 'OPTIMISM_RPC', 'ARBITRUM_RPC', 'GNOSIS_RPC', 'BSC_RPC'];
```
Comment (0xGorilla) [inline solidity/scripts/MultichainDeploy.sol:14]: It seems you deployed to: 
Ethereum Mainnet
Optimistic Ethereum
GnosisScan
Goerli Testnet
Arbitrum
Arb Goerli
Goerli
BNB
Mumbai

Nevertheless, there are only 6 chains in this array, any reason for that?

File: .gitignore [hunk 4]
```diff
@@ -18,4 +18,5 @@ out/*/*
 !out/Greeter.sol/*
 
 # Keep the latest deployment only
-broadcast/*/*/*
+broadcast/**
+!broadcast/**/run-latest.json
```
Comment (0xShaito) [inline .gitignore:22]: Do we really want to push all of this runs?

File: solidity/scripts/MultichainDeploy.sol (hunk 3 above)
Comment (0xShaito) [inline solidity/scripts/MultichainDeploy.sol:14]: Can you add the rpcs that are missing in the .env.example?

**Review by 0xShaito — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (prathmeshkhandelwal1) [reply .gitignore:22]: Yes, we want to keep track of all the deployments, Tagging @just-a-node here for clarity.

File: solidity/scripts/MultichainDeploy.sol (hunk 3 above)
Comment (prathmeshkhandelwal1) [reply solidity/scripts/MultichainDeploy.sol:14]: Sure, Thanks for suggestion

**Review by 0xShaito — APPROVED** [review]

File: .gitignore (hunk 4 above)
Comment (just-a-node) [reply .gitignore:22]: Yeah, mainly for proof of deployment and contract address. Is there a concern in exposing the broadcasts?

**Review by 0xGorilla — CHANGES_REQUESTED** [review]

File: solidity/scripts/MultichainDeploy.sol (hunk 2 above)
Comment (just-a-node) [reply solidity/scripts/MultichainDeploy.sol:18]: Alphanum random string in Python

File: solidity/scripts/MultichainDeploy.sol (hunk 3 above)
Comment (just-a-node) [reply solidity/scripts/MultichainDeploy.sol:14]: Testnet and mainnet deployments were done in 2 different runs of the deploy script, what's in here was the remnants of the mainnet deployment. Total deployments should be 10 in total.

Mainnets ([commit](https://github.com/prathmeshkhandelwal1/xERC20/commit/41c5dc8094ce1edba2d5b6b8cd903e05e6ab4174)):
- Ethereum mainnet
- Optimism 
- Gnosis
- Polygon
- Arbitrum
- BNB

Testnets ([commit](https://github.com/prathmeshkhandelwal1/xERC20/commit/954f8efee1d174f087ccf19864b8f9421241fae2)):
- Goerli 
- Optimism Goerli
- Arbitrum Goerli
- Mumbai

**Review by 0xGorilla — APPROVED** [review]

**Pull request merged**